fuel/datasets/base.py

```diff
@@ -263,73 +262,76 @@ def load(self):
         """
         pass
 
-    @staticmethod
-    def lazy_properties(*lazy_properties):
-        r"""Decorator to assign lazy properties.
 
-        Used to assign "lazy properties" on :class:`InMemoryDataset`
-        classes.  Please see the documentation there for a discussion on
-        what lazy properties are and why they are needed.
+def do_not_pickle_properties(*lazy_properties):
```

Comment:
I would say `do_not_pickle_attributes` because for the user they are attributes, it's only through the internals of this decorator that they are made into properties.